torch/csrc/autograd/python_variable.cpp

```diff
@@ -364,6 +364,54 @@ static PyObject* THPVariable_make_subclass(PyObject* _ignored, PyObject* args, P
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject* THPVariable_make_wrapper_subclass(PyObject* _ignored, PyObject* args, PyObject* kwargs) {
+  HANDLE_TH_ERRORS
+  // NB: pin_memory doesn't actually do anything
+  // TODO: strides variant?
+  static PythonArgParser parser({
+    "_make_wrapper_subclass(PyObject* cls, IntArrayRef size, *, MemoryFormat? memory_format=None, ScalarType dtype=None, Layout layout=torch.strided, Device device=None, bool pin_memory=False, bool requires_grad=False)",
+  });
+  ParsedArgs<8> parsed_args{};
+  auto r = parser.parse(args, kwargs, parsed_args);
+  // TODO: handle has_torch_function, maybe?
+  PyObject* cls = r.pyobject(0);
+  if (!PyType_Check(cls)) {
+    throw torch::TypeError("cls must be a type (got %s)", Py_TYPE(cls)->tp_name);
+  }
+
+  // This is an important safety check; without it, the default behavior will be
+  // to continue on to the underlying CPU/CUDA kernel advertised by the dispatch
+  // key, which will immediately segfault because the data pointer is null.  By
+  // forcing users to define __torch_dispatch__ we ensure this does not happen
+  if (PyObject_FastGetAttrString(cls, "__torch_dispatch__").ptr() == nullptr) {
+    throw torch::TypeError("%s must define __torch_dispatch__", ((PyTypeObject*)cls)->tp_name);
```

Comment:
TORCH_CHECK

Reply:
Is direct TypeError actually deprecated? There are still a lot of examples of this, some within this file.

Reply:
Ho, `TORCH_CHECK_TYPE()` to get a TypeError would work.
I think we want to make all of them consistent no?